code/game/objects/structures/table_frames.dm

```diff
@@ -30,21 +36,44 @@
 	if(!istype(stack))
 		return FALSE
 
-	if(!stack.table_type)
+	var/obj/structure/table/new_table_type = stack.table_type
+	if(length(restrict_table_types))
+		var/valid_stack_type = FALSE
+		for(var/obj/item/stack/current_stack as anything in restrict_table_types)
+			if(istype(stack, current_stack))
+				if(restrict_table_types[current_stack])
```

Comment:
this if check seems redundant? We know that `current_stack` must be an entry in `restrict_table_types` at this point b/c of the for loop.